From 092c7dfb3e36d43ede56bd76b62a9c450b0d78f0 Mon Sep 17 00:00:00 2001
From: Jan Janssen <jan-janssen@users.noreply.github.com>
Date: Tue, 29 Jul 2025 11:01:05 +0200
Subject: [PATCH] Fix parsing of line kpoints in vasp 6.4.3 (#70)

* Fix Vasp 6.4.3 kpoints in line

* fix

* [pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

---------

Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>
---
 pyiron_vasp/vasp/parser/outcar.py | 28 +++++++++++++++++++++++-----
 1 file changed, 23 insertions(+), 5 deletions(-)

diff --git a/pyiron_vasp/vasp/parser/outcar.py b/pyiron_vasp/vasp/parser/outcar.py
index 537b292..ee8812e 100644
--- a/pyiron_vasp/vasp/parser/outcar.py
+++ b/pyiron_vasp/vasp/parser/outcar.py
@@ -326,22 +326,40 @@ def get_irreducible_kpoints(
         weight_lst = []
         planewaves_lst = []
         trigger_number_str = "Subroutine IBZKPT returns following result:"
+        trigger_number_str_alt = "k-points in reciprocal lattice and weights:"
+        trigger_number_str_total = (
+            "position of ions in fractional coordinates (direct lattice)"
+        )
         trigger_plane_waves_str = "k-point  1 :"
+        trigger_plane_waves_alt_str = "k-point     1 :"
         trigger_number = 0
+        trigger_number_alt = 0
+        trigger_number_alt_total = 0
         trigger_plane_waves = 0
         lines = _get_lines_from_file(filename=filename, lines=lines)
         for i, line in enumerate(lines):
             line = line.strip()
             if trigger_number_str in line:
                 trigger_number = int(i)
+            elif trigger_number_str_alt in line:
+                trigger_number_alt = int(i) + 1
+            elif trigger_number_alt != 0 and trigger_number_str_total in line:
+                trigger_number_alt_total = int(i) - 1
             elif planewaves:
-                if trigger_plane_waves_str in line:
+                if (
+                    trigger_plane_waves_str in line
+                    or trigger_plane_waves_alt_str in line
+                ) and "plane waves: " in line:
                     trigger_plane_waves = int(i)
-        number_irr_kpoints = int(lines[trigger_number + 3].split()[1])
-        if reciprocal:
-            trigger_start = trigger_number + 7
+        if trigger_number != 0:
+            number_irr_kpoints = int(lines[trigger_number + 3].split()[1])
+            if reciprocal:
+                trigger_start = trigger_number + 7
+            else:
+                trigger_start = trigger_number + 10 + number_irr_kpoints
         else:
-            trigger_start = trigger_number + 10 + number_irr_kpoints
+            trigger_start = trigger_number_alt
+            number_irr_kpoints = trigger_number_alt_total - trigger_number_alt
         for line in lines[trigger_start : trigger_start + number_irr_kpoints]:
             line = line.strip()
             line = _clean_line(line)
